Assets/Scripts/Utility/AssetInjection/WorldDataReplacement.cs

```diff
@@ -86,15 +92,234 @@ public static string GetBuildingReplacementFilename(string blockName, int blockI
+        public static bool GetDFLocationReplacementData(int regionIndex, int locationIndex, out DFLocation dfLocation)
+        {
+            if (DaggerfallUnity.Settings.AssetInjection)
+            {
+                // If found, return a previously cached DFLocation
+                int locationKey = MakeLocationKey(regionIndex, locationIndex);
+                if (locations.ContainsKey(locationKey))
+                {
+                    dfLocation = locations[locationKey];
+                    return dfLocation.LocationIndex != noReplacementLocation.LocationIndex;
+                }
+
+                string fileName = GetDFLocationReplacementFilename(regionIndex, locationIndex);
+                TextAsset locationReplacementJsonAsset;
+
+                // Seek from loose files
+                if (File.Exists(Path.Combine(worldDataPath, fileName)))
+                {
+                    string locationReplacementJson = File.ReadAllText(Path.Combine(worldDataPath, fileName));
+                    dfLocation = (DFLocation)SaveLoadManager.Deserialize(typeof(DFLocation), locationReplacementJson);
+                }
+                // Seek from mods
+                else if (ModManager.Instance != null && ModManager.Instance.TryGetAsset(fileName, false, out locationReplacementJsonAsset))
+                {
+                    dfLocation = (DFLocation)SaveLoadManager.Deserialize(typeof(DFLocation), locationReplacementJsonAsset.text);
+                }
+                else
+                {
+#if !UNITY_EDITOR // Cache that there's no replacement location data, so only look for replaced locations once (unless running in editor)
+                    locations[locationKey] = noReplacementLocation;
+#endif
+                    dfLocation = noReplacementLocation;
+                    return false;
+                }
+                // Assign any new blocks in this location a block index if they haven't already been assigned
+                if (AssignBlockIndices(dfLocation))
+                {
+#if !UNITY_EDITOR   // Cache location data for replaced locations if new blocks have been assigned indices (unless running in editor)
+                    locations[locationKey] = dfLocation;
+#endif
+                }
+                Debug.LogFormat("Found DFLocation override, region:{0}, index:{1}", regionIndex, locationIndex);
+                return true;
+            }
+            dfLocation = noReplacementLocation;
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the index for a new added block
+        /// </summary>
+        /// <param name="blockName">Block name</param>
+        /// <returns>Block index, or -1 if not found</returns>
+        public static int GetNewDFBlockIndex(string blockName)
+        {
+            if (newBlockIndices.ContainsKey(blockName))
+                return newBlockIndices[blockName];
+            else
+                return -1;
+        }
```

Comment:
I'd suggest `TryGetValue` for performance reasons:

```cs
int index;
return newBlockIndices.TryGetValue(blockName, out index) ? index : -1;
```

Reply:
I use both forms depending on what is clearer to read. I prefer easy to read over saving a single hash lookup.